src/pkg/variables/common.go

```diff
@@ -0,0 +1,31 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2021-Present The Zarf Authors
+
+// Package variables contains functions for interacting with variables
+package variables
+
+import "log/slog"
+
+// VariableConfig represents a value to be templated into a text file.
+type VariableConfig struct {
+	templatePrefix string
+	deprecatedKeys map[string]string
+
+	ApplicationTemplates map[string]*TextTemplate
+	SetVariableMap       SetVariableMap
+	Constants            []Constant
+
+	logger *slog.Logger
+}
+
+// New creates a new VariableConfig
+func New(templatePrefix string, deprecatedKeys map[string]string, setVariableMap SetVariableMap, constants []Constant, logger *slog.Logger) *VariableConfig {
+	return &VariableConfig{
+		templatePrefix:       templatePrefix,
+		deprecatedKeys:       deprecatedKeys,
+		ApplicationTemplates: make(map[string]*TextTemplate),
+		SetVariableMap:       setVariableMap,
+		Constants:            constants,
+		logger:               logger,
+	}
+}
```

Comment:
I would prefer these public properties be private and have their own receivers for setters / getters / append.

I think this will help w/ readability so we can more clearly see places where these maps and slices are getting _initialized_, _overrwitten_, and _appended_ to.

Right now `constants` is initialized in `New`, but is only ever initialized as empty, and overwritten in certain portions.  Its a little weird to me that something labeled as "constant" is being used as a mutable object xD

Reply:
this should be fixed - lemme know if you have more feedback